src/test/java/automationpractice/com/pages/ProductDetailsPage.java

```diff
@@ -0,0 +1,251 @@
+    private WebElementFacade changeColorToWhite;
+
+    @FindBy(id = "color_11")
+    private WebElementFacade changeColorToBlack;
+
+    @FindBy(id = "color_15")
+    private WebElementFacade changeColorToGreen;
+
+    @FindBy(id = "color_16")
+    private WebElementFacade changeColorToYellow;
+
+    @FindBy(className = "btn-twitter")
+    private WebElementFacade clickOnTwitterButton;
+
+    @FindBy(className = "btn-facebook")
+    private WebElementFacade clickOnFacebookButton;
+
+    @FindBy(className = "btn-google-plus")
+    private WebElementFacade clickOnGooglePlusButton;
+
+    @FindBy(className = "btn-pinterest")
+    private WebElementFacade clickOnPinteresButton;
+
+    private int currentQuantityInProductDetailsPage;
+
+    private String currentSizeInProductDetailsPage;
+
+    @FindBy(xpath = "//*[@id = 'color_to_pick_list']//*[contains(@id,'color')]")
+    private List<WebElementFacade> listOfColorsAndSizes;
+
+    private String currentColorInProductDetailsPage;
+
+    public ProductDetailsPage() {
+    }
+
+    public void clickOnAddToCartButton() {
+        clickOnAddToCartButton.click();
+    }
+
+    public void clickProceedToCheckoutButton() {
+        clickProceedToCheckoutButton.click();
+    }
+
+    private void clickPlusQuantityButtonOnProductDetailsPage() {
+        clickPlusQuantityButtonOnProductDetailsPage.click();
+    }
+
+    private void clickMinusQuantityButtonOnProductDetailsPage() {
+        clickMinusQuantityButtonOnProductDetailsPage.click();
+    }
+
+    public int clickPlusAndMinusButtonGivenNumberOfTimesOnProductDetailsPage(int plus, int minus) {
+        currentQuantityInProductDetailsPage = 1;
+        for (int i = 0; i < plus; i++) {
+            waitABit(1000);
+            clickPlusQuantityButtonOnProductDetailsPage();
+            currentQuantityInProductDetailsPage++;
+        }
+        for (int i = 0; i < minus; i++) {
+            waitABit(1000);
+            clickMinusQuantityButtonOnProductDetailsPage();
+            currentQuantityInProductDetailsPage--;
+        }
+        return currentQuantityInProductDetailsPage;
+    }
+
+    public int getCurrentQuantitySetUpInProductDetailsPage() {
+        return currentQuantityInProductDetailsPage;
+    }
+
+    public String changeSizeToGivenSizeInProductDetailsPage(String sizeWrittenInLargeLetters) {
+        currentSizeInProductDetailsPage = sizeWrittenInLargeLetters;
+        switch (sizeWrittenInLargeLetters) {
+            case "S":
+                selectSSize.click();
+                break;
+            case "M":
+                selectMSize.click();
+                break;
+            case "L":
+                selectLSize.click();
+                break;
+            default:
+                selectSSize.click();
+                break;
+        }
+        return sizeWrittenInLargeLetters;
+    }
+
+    public String getCurrentSizeSetUpInProductDetailsPage() {
+        return currentSizeInProductDetailsPage;
+    }
```

Comment:
Jeśli po tym jak kliknąłeś rozmiar nic sie nie wydarzy i rozmiar się nie zmieni to i tak jako zaznaczony rozmiar dostaniesz ten na który chciałeś zmienić. Test przejdzie nawet gdyby tu był jakiś bug. powinieneś wyciągnąc wybrany rozmiar z webElementu